### PR TITLE
Add self-admitted technical debt calculation

### PR DESCRIPTION
**Pull request by suhaibmujahid**
This is a pull request to calculate the Technical Debt matrix based on the developers' comments in their source code.

I have added the matrix to the quality result but still not calculated in the final value of the quality measurement.

### CODE REVIEW

**Comment by satazor** [comment]
This looks cool! Will review next weekend, out of time atm.

**Comment by suhaibtamimi** [comment]
Hey guys,
Do you have a chance to look into this!

Have a good weekend :)

File: lib/analyze/evaluate/quality.js [hunk 1]
```diff
@@ -160,6 +160,21 @@ function evaluateBranding(collected) {
...
+ * @param {object} collected The collected information
+ *
+ * @return {number} The technical debt evaluation (from 0 to 1)
+ */
+function evaluateTechnicalDebts(collected) {
+    if (!collected.source) {
+        return 0;
+    }
+
+    return collected.source.technicalDebts
```
Comment (satazor) [inline lib/analyze/evaluate/quality.js]: Can you add a `;` at the end?

**Review by satazor — CHANGES_REQUESTED** [review]
Additionally, you are missing adding the `technicalDebts` to the score calculation. Do not forget that the more the debts, the lower the score should be. Probably the evaluation should reflect this.. perhaps should be `1 - debts.percent`.

File: lib/analyze/evaluate/quality.js (hunk 1 above)
Comment (suhaibtamimi) [inline lib/analyze/evaluate/quality.js]: yes, sure :)

**Comment by satazor** [comment]
@suhaibtamimi Check my other comment above.

**Comment by suhaibtamimi** [comment]
@satazor The requested changes are ready.

**Review by satazor — APPROVED** [review]

**Comment by satazor** [comment]
@suhaibtamimi I think this is almost ready! I would just like to request one further change because I think it makes sense: instead of having `technicalDebts` in evaluation & score, what if we had `code`? This way we could group code related evaluation together, just like we do with `carefulness`, `health`, etc. For instance, `code` could evaluate technical debts, code complexity, and other code related metrics. Makes sense?

What do you think @atduarte ?

**Comment by suhaibtamimi** [comment]
I agree it makes sense; I did move the `technicalDebts` evaluation under `code` category.

**Pull request closed without merging**